src/tcp/stream.rs

```diff
@@ -346,33 +360,58 @@ fn choose_connections(
     all_incoming: BoxStream<TcpStream, SingleRendezvousAttemptError>,
     our_pk: PublicKey,
     their_pk: PublicKey,
+    our_sk: SecretKey,
 ) -> BoxStream<TcpStream, SingleRendezvousAttemptError> {
-    const CHOOSE: [u8; 6] = [b'c', b'h', b'o', b'o', b's', b'e'];
+    let encrypted_msg = unwrap!(secure_serialise(&CHOOSE, &their_pk, &our_sk));
```

Comment:
why are we using unwarps in production ?

Reply:
no rational reason, I guess I was just going autopilot, will fix this.